src/java.desktop/windows/native/libawt/windows/awt_Window.cpp

```diff
@@ -1406,25 +1406,17 @@ BOOL AwtWindow::UpdateInsets(jobject insets)
     } else {
         m_insets.top = -1;
     }
+
     if (m_insets.left < 0 || m_insets.top < 0 ||
-        m_insets.right < 0 || m_insets.bottom < 0)
-    {
+        m_insets.right < 0 || m_insets.bottom < 0) {
         /* This window hasn't been sized yet -- use system metrics. */
         jobject target = GetTarget(env);
         if (IsUndecorated() == FALSE) {
             /* Get outer frame sizes. */
-            LONG style = GetStyle();
-            if (style & WS_THICKFRAME) {
-                m_insets.left = m_insets.right =
-                    ::GetSystemMetrics(SM_CXSIZEFRAME) + extraPaddedBorderInsets;
-                m_insets.top = m_insets.bottom =
-                    ::GetSystemMetrics(SM_CYSIZEFRAME) + extraPaddedBorderInsets;
-            } else {
-                m_insets.left = m_insets.right =
-                    ::GetSystemMetrics(SM_CXDLGFRAME) + extraPaddedBorderInsets;
-                m_insets.top = m_insets.bottom =
-                    ::GetSystemMetrics(SM_CYDLGFRAME) + extraPaddedBorderInsets;
-            }
+            m_insets.left = m_insets.right =
+                ::GetSystemMetrics(SM_CXSIZEFRAME) + extraPaddedBorderInsets;
+            m_insets.top = m_insets.bottom =
+                ::GetSystemMetrics(SM_CYSIZEFRAME) + extraPaddedBorderInsets;
```

Comment:
Is it worth adding a comment clarifying why the style isn't checked? Maybe not, since now it's gone from the code. On the other hand, a reader may wonder why we always uses the sizing frame insets even for non-resizeable frames.

Reply:
@aivanov-jdk I agree, comment would add more clarity. Additionally instead of using **SM_CXSIZEFRAME** (the name can be ambiguous too) can we change it to **SM_CXFRAME** ? Since both are mentioned to be same metrics in the documentation https://learn.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-getsystemmetrics

Reply:
> Additionally instead of using **SM_CXSIZEFRAME** (the name can be ambiguous too) can we change it to **SM_CXFRAME** ?

Yes, `SM_CXFRAME` is neutral, it doesn't refer to either resizeable or non-resizeable frame.